pennylane/estimator/ops/qubit/non_parametric_ops.py

```diff
@@ -0,0 +1,884 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Resource operators for non parametric single qubit operations."""
+
+from pennylane.estimator.ops.identity import Identity
+from pennylane.estimator.resource_operator import (
+    CompressedResourceOp,
+    GateCount,
+    ResourceOperator,
+    resource_rep,
+)
+from pennylane.exceptions import ResourcesUndefinedError
+
+# pylint: disable=arguments-differ
+
+
+class Hadamard(ResourceOperator):
+    r"""Resource class for the Hadamard gate.
+
+    Args:
+        wires (Sequence[int] or int, optional): the wire the operation acts on
+
+    Resources:
+        The Hadamard gate is treated as a fundamental gate and thus it cannot be decomposed
+        further. Requesting the resources of this gate raises a :class:`~.pennylane.exceptions.ResourcesUndefinedError` error.
+
+    .. seealso:: The corresponding PennyLane operation :class:`~.pennylane.Hadamard`.
+
+    """
+
+    num_wires = 1
+
+    @classmethod
+    def resource_decomp(cls, **kwargs) -> list[GateCount]:
+        r"""Returns a list representing the resources of the operator. Each object represents a quantum gate
+        and the number of times it occurs in the decomposition.
+
+        Resources:
+            The ``Hadamard`` gate is treated as a fundamental gate and thus it cannot be decomposed
+            further. Requesting the resources of this gate raises a :class:`~.pennylane.exceptions.ResourcesUndefinedError` error.
+
+        Raises:
+            ResourcesUndefinedError: This gate is fundamental, no further decomposition defined.
+        """
+        raise ResourcesUndefinedError
```

Comment:
Wait, why are we overriding this if it just has the base behavior? Couldn't we just leave it be?